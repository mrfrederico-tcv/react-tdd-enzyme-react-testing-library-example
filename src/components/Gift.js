import React, { Component } from 'react'
import {
  Form,
  FormGroup,
  FormControl,
  FormLabel,
  Button,
} from 'react-bootstrap'
import styled from 'styled-components'

const Root = styled.div``

class App extends Component {
  state = {
    person: '',
    present: '',
  }

  handleInputPersonChange = evt => {
    const { value } = evt.target
    this.setState({ person: value })
  }

  handleInputPresentChange = evt => {
    const { value } = evt.target
    this.setState({ present: value })
  }

  handleRemoveGift = () => {
    const { onRemoveGift, gift } = this.props
    onRemoveGift(gift.id)
  }

  render() {
    return (
      <Root>
        <Form>
          <FormGroup>
            <FormLabel>Person</FormLabel>
            <FormControl
              testId="InputPerson"
              onChange={this.handleInputPersonChange}
            />
          </FormGroup>
          <FormGroup>
            <FormLabel>Present</FormLabel>
            <FormControl
              testId="InputPresent"
              onChange={this.handleInputPresentChange}
            />
          </FormGroup>
        </Form>
        <Button testId="RemoveButton" onClick={this.handleRemoveGift}>
          Remove Gift
        </Button>
      </Root>
    )
  }
}

export default App
