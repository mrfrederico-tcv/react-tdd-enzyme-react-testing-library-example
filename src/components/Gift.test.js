import React from 'react'
import { shallow } from 'enzyme'
import Gift from './Gift'

describe('Gift', () => {
  const ID = 1

  const mockOnRemove = jest.fn()
  const props = { gift: { id: ID }, onRemoveGift: mockOnRemove }
  const gift = shallow(<Gift {...props} />)

  it('render correctly', () => {
    expect(gift).toMatchSnapshot()
  })

  it('init a person and present in `state`', () => {
    const state = gift.state()
    expect(state).toEqual({ person: '', present: '' })
  })

  describe('when typing into the person input', () => {
    const PERSON = 'UNCLE'

    beforeEach(() => {
      gift
        .find('[testId="InputPerson"]')
        .simulate('change', { target: { value: PERSON } })
    })

    it('updates the person in `state`', () => {
      const { person } = gift.state()
      expect(person).toEqual(PERSON)
    })
  })

  describe('when typing into the present input', () => {
    const PRESENT = 'Golf Clubs'

    beforeEach(() => {
      gift
        .find('[testId="InputPresent"]')
        .simulate('change', { target: { value: PRESENT } })
    })

    it('updates the present in `state`', () => {
      const { present } = gift.state()
      expect(present).toEqual(PRESENT)
    })
  })

  describe('when clicking the `Remove Gift` button', () => {
    beforeEach(() => {
      gift.find('[testId="RemoveButton"]').simulate('click')
    })

    it('calls the handleRemoveGift callback', () => {
      expect(mockOnRemove).toHaveBeenCalledWith(ID)
    })
  })
})
