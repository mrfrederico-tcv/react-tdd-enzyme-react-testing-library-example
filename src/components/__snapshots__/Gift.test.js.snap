// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift render correctly 1`] = `
<styled.div>
  <Form
    inline={false}
  >
    <FormGroup>
      <FormLabel
        column={false}
        srOnly={false}
      >
        Person
      </FormLabel>
      <FormControl
        onChange={[Function]}
        testId="InputPerson"
      />
    </FormGroup>
    <FormGroup>
      <FormLabel
        column={false}
        srOnly={false}
      >
        Present
      </FormLabel>
      <FormControl
        onChange={[Function]}
        testId="InputPresent"
      />
    </FormGroup>
  </Form>
  <Button
    active={false}
    disabled={false}
    onClick={[Function]}
    testId="RemoveButton"
    type="button"
    variant="primary"
  >
    Remove Gift
  </Button>
</styled.div>
`;
